src/components/shared/chart/Viewport.js

```diff
@@ -523,6 +527,14 @@ export const withChartViewport: WithChartViewport<*, *> =
         window.removeEventListener('mouseup', this._mouseUpListener, true);
       }
 
+      componentDidUpdate(prevProps: ViewportProps) {
+        if (
+          prevProps.panelLayoutGeneration !== this.props.panelLayoutGeneration
+        ) {
+          this._setSize();
```

Comment:
Did you try to use `_setSizeNextFrame` ? Not sure this makes a real difference though.

Reply:
Yeah, that's what I started with, but it ends up flashing the screen with a white background. We should re-evaluate the logic of the `_setSizeNextFrame` in this compenent, as it was originally addressing some ye olde timeline panel (stack chart) problems.